numpyro/infer/reparam.py

```diff
@@ -127,14 +120,15 @@ class TransformReparam(Reparam):
     """
     def __call__(self, name, fn, obs):
         assert obs is None, "TransformReparam does not support observe statements"
-        fn, batch_shape = self._unexpand(fn)
+        fn, batch_shape, event_dim = self._unwrap(fn)
         assert isinstance(fn, dist.TransformedDistribution)
 
         # Draw noise from the base distribution.
-        # We need to make sure that we have the same batch_shape
-        reinterpreted_batch_ndims = fn.event_dim - fn.base_dist.event_dim
+        base_event_dim = event_dim
+        for t in reversed(fn.transforms):
+            base_event_dim += t.domain.event_dim - t.codomain.event_dim
```

Comment:
This is important in cases like
```py
TransformedDistribution(Normal(np.zeros(10), 1),  biject_to(constraits.simplex))
```
where `fn.base_dist.event_dim == 0` but `fn.event_dim == 1`.

Reply:
I think we need to use the logic that we recently added to ComposedTransform here.
```python
base_event_dim = event_dim
t = ComposedTransform(fn.transforms)
base_event_dim += t.domain.event_dim - t.codomain.event_dim
```

Reply:
Hmm, it seems to not needed because going backward, event_dim is always guaranteed to greater than `domain.event_dim`. Please ignore my comment.